apps/main/src/dex/components/PagePool/Swap/index.tsx

```diff
@@ -88,15 +86,32 @@ const Swap = ({
 
   const poolId = poolData?.pool?.id
   const haveSigner = !!signerAddress
+
   const userFromBalance = userPoolBalances?.[formValues.fromAddress]
   const userToBalance = userPoolBalances?.[formValues.toAddress]
+
   const fromUsdRate = usdRatesMapper[formValues.fromAddress]
   const toUsdRate = usdRatesMapper[formValues.toAddress]
 
-  const { selectList, swapTokensMapper } = useMemo(
-    () => getSwapTokens(tokensMapper, poolDataCacheOrApi),
-    [poolDataCacheOrApi, tokensMapper],
-  )
+  const { selectList, swapTokensMapper } = useMemo(() => {
+    const { selectList, swapTokensMapper } = getSwapTokens(tokensMapper, poolDataCacheOrApi)
+
+    return {
+      selectList: selectList
+        .filter((token) => !!token)
+        .map((token) => ({
+          chain: network?.networkId ?? '',
```

Comment:
if chain is optional it should be optional not an empty string

Reply:
TokenIcon chain used be required, but I've now made it optional and default to `''` inside the component itself